src/openzaak/components/zaken/api/serializers/zaken.py

```diff
@@ -1720,6 +1727,52 @@ def update(self, instance, validated_data):
         }
 
 
+class ZaakAfsluitenSerializer(ConvenienceSerializer):
+    zaak = ZaakSubSerializer(required=True, partial=True)
+    resultaat = ResultaatSubSerializer(required=True)
+    status = StatusSubSerializer(required=True)
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        zaak_serializer = ZaakSerializer(
+            instance=instance,
+            data=self.initial_data.get("zaak", {}),
+            partial=True,
+            context=self.context,
+        )
+        zaak_serializer.is_valid(raise_exception=True)
+        zaak = zaak_serializer.save()
+
+        zaak_data = {"zaak": zaak.get_absolute_api_url(request=self.context["request"])}
+
+        resultaat_serializer = ResultaatSerializer(
+            data={**self.initial_data.get("resultaat", {}), **zaak_data},
+            context=self.context,
+        )
+        resultaat_serializer.is_valid(raise_exception=True)
+        resultaat = resultaat_serializer.save()
+
+        if not zaak.einddatum:
+            zaak.einddatum = now().date()
+            zaak.save(update_fields=["einddatum"])
```

Comment:
`einddatum` will be set by the `StatusSerializer` (based on `datum_status_gezet`), so I think this can be removed
```suggestion
```